### PR TITLE
Users can complete a Transfer project

### PR DESCRIPTION
**Pull request by lozette**
## Changes

A Transfer project can be completed if its transfer date is confirmed and in the past. 

This PR includes some refactoring to allow for Conversion and Transfer projects to be completed under different sets of conditions.

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by lozette** [comment]
[Users could complete a transfer](https://trello.com/c/3f1ge07K/1916-users-could-complete-a-transfer)

File: app/controllers/projects_complete_controller.rb [hunk 1]
```diff
@@ -3,21 +3,16 @@ def complete
...
     authorize @project, :update?
 
-    if project_completable?
+    if @project.completable?
       set_project_completed_at
 
       render :completed
     else
-      flash[:alert] = I18n.t("project.complete.unable_to_complete_html")
+      flash[:alert] = I18n.t("#{@project.type.downcase.gsub("::", "_")}.complete.unable_to_complete_html")
```
Comment (mec) [inline app/controllers/projects_complete_controller.rb]: `@project.type.parameterize(separator: "_")` might give you what you want here?

**Review by mec — APPROVED** [review]
🚢

File: app/controllers/projects_complete_controller.rb (hunk 1 above)
Comment (lozette) [reply app/controllers/projects_complete_controller.rb]: Ohhhh good tip

**Pull request merged**